### PR TITLE
Avoid shapeless conflicts in spark doc

### PR DESCRIPTION
**Pull request by rberenguel**
A very trivial doc to explain how to use shading with `sbt-assembly` (since I guess it's happening to more people and seems hard to Google for). I tried to keep the overall style in the section titles, but I'm not sure though that this warrants a different file, maybe it could be under some of the other headings.

### CODE REVIEW

**Comment by melrief** [comment]
Thanks @rberenguel for the PR! PureConfig uses [`tut`](https://github.com/tpolecat/tut) to generate the documentation so that we can validate it. Canyou move your new section inside the [documentation source](https://github.com/pureconfig/pureconfig/tree/master/docs/src/main/tut) and then build the documentation?

@derekmorr do you have time to have a look at this too?

File: README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ A boilerplate-free Scala library for loading configuration files.
 - [Handling missing keys](docs/handling-missing-keys.md)
 - [Example](docs/example.md)
 - [Whence the config files](docs/whence-the-config-files.md)
+- [Playing nice with Spark's Shapeless](docs/playing-nice-with-sparks-shapeless.md)
```
Comment (derekmorr) [inline README.md]: This should also be added to `core/src/main/tut/README.md` or it will be overwritten when the docs are next generated.

**Review by derekmorr — COMMENTED** [review]
Since these changes are self-contained to the main README, I think they just need to go in `core/src/main/tut/README.md`. Tut can't check the SBT DSL, so there's no need to use the `tut` annotation on the examples.

File: README.md (hunk 1 above)
Comment (rberenguel) [reply README.md]: @derekmorr I think you were commenting while I was committing :), but I'm not sure (I moved this to a FAQ section in the main `README.md`)

Comment (rberenguel) [reply README.md]: Oh, sorry, missed the `core` in your comment. Fixing the real "main" `README.md`

**Review by melrief — APPROVED** [review]

**Review by derekmorr — APPROVED** [review]

**Pull request merged**